ui/pages/confirmation/confirmation.js

```diff
@@ -256,20 +260,37 @@ export default function ConfirmationPage({
     // viewed index, reset the index.
     if (
       pendingConfirmations.length === 0 &&
+      pendingApprovalFlows.length === 0 &&
       redirectToHomeOnZeroConfirmations
     ) {
       history.push(DEFAULT_ROUTE);
-    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+    } else if (
+      pendingConfirmations.length &&
+      pendingConfirmations.length <= currentPendingConfirmation
+    ) {
       setCurrentPendingConfirmation(pendingConfirmations.length - 1);
     }
   }, [
     pendingConfirmations,
+    pendingApprovalFlows,
     history,
     currentPendingConfirmation,
     redirectToHomeOnZeroConfirmations,
   ]);
 
+  useEffect(() => {
+    setApprovalFlowLoadingText(
+      pendingApprovalFlows.length > 0
+        ? pendingApprovalFlows[pendingApprovalFlows.length - 1].loadingText
+        : null,
+    );
+  }, [pendingApprovalFlows]);
+
   if (!pendingConfirmation) {
+    if (pendingApprovalFlows.length > 0) {
+      return <Loading loadingMessage={t(approvalFlowLoadingText)} />;
```

Comment:
Not needed now of course but in future we could instead render our own `FlowLoadingPage` component to encapsulate any other loading UI we need.

Reply:
Yes, I'd probably leave this one until we have a specific design and add features, then worth creating our own component.